### PR TITLE
add script to audit nupkg during build

### PR DESCRIPTION
**Pull request by MS-TimothyMothra**
adds a script to audit nupkgs during build

### CODE REVIEW

File: NugetAudit.ps1 [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    $id = $nuspecXml.package.metadata.id;
+
+    $message = "Package Id: $id";
+    $requirement = "Must begin with 'Microsoft.'"
+
+    Test-Condition ($id.StartsWith("Microsoft.")) $message $requirement;
+}
+
+function Get-IsValidVersioning() {
+    #TODO
```
Comment (cijothomas) [inline NugetAudit.ps1]: reminder about todos.

File: NugetAudit.ps1 [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    $message = "Require License Acceptance: $requireLicenseAcceptance";
+    $requirement = "Not mandatory requirement."
+    $recommendation = "Should require.";
+
+    Test-MultiCondition ($true) ($requireLicenseAcceptance -eq $true) $message $requirement $recommendation;
+}
+
+function Get-IsValidCopyright([xml]$nuspecXml) {
+    $copyright = $nuspecXml.package.metadata.copyright;
+    $requiredCopyright = "$([char]0x00A9) Microsoft Corporation. All rights reserved.";#"© Microsoft Corporation. All rights reserved.";
```
Comment (cijothomas) [inline NugetAudit.ps1]: minor - move all the expected/required values to the top of this file to make it easier to read later.

File: NugetAudit.ps1 [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+        Get-IsValidPackageId $nuspecXml;
+        Get-IsValidAuthors $nuspecXml;
+        Get-IsValidOwners $nuspecXml;
+        Get-IsValidProjectUrl $nuspecXml;
+        Get-IsValidLicenseUrl $nuspecXml;
+        Get-IsValidLicenseAcceptance $nuspecXml;
+        Get-IsValidCopyright $nuspecXml;
+        Get-IsValidLogoUrl $nuspecXml $unzipPath;
+        Get-IsValidDescription $nuspecXml;
+        Get-IsValidTags $nuspecXml;
```
Comment (cijothomas) [inline NugetAudit.ps1]: suggestion - check against a list of tags we should be having.

File: NugetAudit.ps1 [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+function Get-IsValidVersioning() {
+    #TODO
+}
+
+function Get-IsValidAuthors([xml]$nuspecXml) {
+    $authors = $nuspecXml.package.metadata.authors;
+
+    $message = "Authors: $authors";
+    $requirement = "Microsoft must be the only owner.";
```
Comment (Dmitry-Matveev) [inline NugetAudit.ps1]: I think you meant to say "the only author", not "the only owner". right?

**Pull request merged**